pyuvsim/telescope.py

```diff
@@ -485,28 +505,55 @@ def _str_to_obj(self, beam_model, use_shared_mem=False):
                 "pip install pyuvsim[sim] or pip install pyuvsim[all] if you also want "
                 "the line_profiler installed.")
 
-        path = beam_model  # beam_model = path to beamfits
+        path = beam_model  # beam_model = path to UVBeam readable file
         uvb = UVBeam()
-        with warnings.catch_warnings():
-            warnings.filterwarnings(
-                "ignore", "The shapes of several attributes will be changing"
-            )
-            if use_shared_mem and (mpi.world_comm is not None):
-                if mpi.rank == 0:
-                    uvb.read_beamfits(path, **self.select_params)
-                    uvb.peak_normalize()
-                for key, attr in uvb.__dict__.items():
-                    if not isinstance(attr, parameter.UVParameter):
-                        continue
-                    if key == '_data_array':
-                        uvb.__dict__[key].value = mpi.shared_mem_bcast(attr.value, root=0)
-                    else:
-                        uvb.__dict__[key].value = mpi.world_comm.bcast(attr.value, root=0)
-                mpi.world_comm.Barrier()
-            else:
-                uvb.read_beamfits(path, **self.select_params)
+        read_kwargs = {}
+        if len(self.select_params) > 0:
+            read_kwargs = self.select_params
+        if len(self.uvb_read_kwargs) > 0 and beam_id in self.uvb_read_kwargs:
+            # do this after select_params to ensure these overwrite select_params
+            read_kwargs.update(self.uvb_read_kwargs[beam_id])
+
```

Comment:
Could be a one-liner: 

```python
read_kwargs = self.select_params | self.uvb_read_kwargs.get(beam_id, {})
```

This only works for python 3.9+, but I think pyuvsim is only working for those versions anyway? If not:

```python
read_kwargs = {**self.select_params, **self.uvb_read_kwargs.get(beam_id, {})}
```